src/Service/Import/GridAffectation/GridAffectationLoader.php

```diff
@@ -193,6 +204,11 @@ public function load(Territory $territory, array $data): void
                         $this->manager->flush();
                     }
                 }
+
+                if (\in_array($partnerType, $ignoreNotifPartnerTypes)) {
+                    $this->manager->flush();
+                    $this->entityManager->getEventManager()->addEventListener(Events::onFlush, $this->userAddedSubscriber);
```

Comment:
Je ne suis pas sûre de comprendre la logique. Pourquoi le premier `$this->manager->flush();` si tu en refais un ici ? Et pourquoi ce addEventListener ? c'est l'inverse du removeEventSubscriber() fait précédemment pour le remettre pour les prochains partenaires ?

Reply:
> Je ne suis pas sûre de comprendre la logique. Pourquoi le premier `$this->manager->flush();` si tu en refais un ici ? Et pourquoi ce addEventListener ? c'est l'inverse du removeEventSubscriber() fait précédemment pour le remettre pour les prochains partenaires ?

L'envoi de mail se fait pendant le flush (qui était jusqu'à présent fait à la toute fin)
Le premier flush ajouté, c'est pour le faire avant la suppression de l'événement, si jamais il y avait des utilisateurs déjà ajoutés.
Le deuxième flush, c'est pour le faire pour les utilisateurs sans notifications, et réactiver l'événement juste après.

Reply:
On pourrait se passait de cette condition et le reajouter apres chaque flush derriere les events listener sont dans un tableau associatif

Reply:
// Overrides listener if a previous one was associated already
            // Prevents duplicate listeners on same event (same instance only)

![image](https://github.com/MTES-MCT/histologe/assets/5757116/42f1f1fb-1bcd-4ad6-9922-922c0ef9d6c7)

Reply:
Ah cool, j'avais peur que ce soit cumulatif ! Ok :)

Reply:
@emilschn 
### On oublie tout ce que j'ai dit sur l'activation et désactivation d’événement

Il y'a un cron qui tourne toutes les semaines qui envoie un mail d'activation de compte aux utilisateurs inactifs

Rajouté un attribut à la classé entité user (ex: enableActivateAccountNotification à true par défaut) => nouvelle migration

Dans le loader mettre à false tout ceux qui ne doivent pas être notifié

Dans le UserCreatedSubscriber ça donnerai
```
if ($entity instanceof User && $entity->getEnableActivateAccountNotification()) {
    $this->sendNotification($entity);
}
````
Mettre à jour `RemindInactiveUserCommand` afin de ne pas envoyer de notification à ceux qui ont false sur enableActivateAccountNotification

Reply:
Ok, c'est fait !